mapper/tedge_mapper/src/converter.rs

```diff
@@ -1,17 +1,43 @@
+use mqtt_client::{Message, Topic, TopicFilter};
+use std::fmt::Display;
+use tracing::error;
+
+#[derive(Debug)]
+pub struct MapperConfig {
+    pub in_topic_filter: TopicFilter,
+    pub out_topic: Topic,
+    pub errors_topic: Topic,
+}
+
 pub trait Converter: Send + Sync {
-    type Error;
+    type Error: Display;
 
-    fn convert(&self, input: &str) -> Result<String, Self::Error>;
+    fn get_mapper_config(&self) -> &MapperConfig;
 
-    fn convert_child_device_payload(
-        &self,
-        _input: &str,
-        _child_id: &str,
-    ) -> Result<String, Self::Error> {
-        Ok("".to_string())
+    fn get_in_topic_filter(&self) -> &TopicFilter {
+        &self.get_mapper_config().in_topic_filter
     }
 
-    fn convert_child_device_creation(&self, _child_id: &str) -> Option<mqtt_client::Message> {
-        None
+    fn convert_messages(&mut self, input: &Message) -> Result<Vec<Message>, Self::Error>;
```

Comment:
Minor: That's a slightly confusing name as the `convert` function is the one that always produces messages, even when there are errors in the conversion. This looks more like a `convert_fallible` if that name makes any sense.

Reply:
A better name could be `try_convert`.